NGRValidator/NGRValidator/NGRRuntime.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  NGRRuntime.swift
+//  NGRValidator
+//
+
+import Foundation
+
+@objc internal final class NGRRuntimeInspector: NSObject {
+    
+    private let model: Any
+    
+    private var mirror: Mirror {
+        return Mirror(reflecting: model)
+    }
+    
+    public init(model: Any) {
+        self.model = model
+    }
+    
+    public func properties() -> [String] {
+        return mirror.properties()
+    }
+    
+    public func value(withKey key: String) -> Any? {
+        return mirror.value(forKey: key)
+    }
+    
+    public func isClassOrStruct() -> Bool {
+        let supportedDisplayStyles: [Mirror.DisplayStyle] = [.struct, .class]
+        
+        return mirror.displayStyle.map { supportedDisplayStyles.contains($0) } ?? false
```

Comment:
Yeah that's better, thanks :)  I've slightly modified your solution because displayStyle is an optional.